zerver/lib/send_email.py

```diff
@@ -106,31 +107,39 @@ def send_email(template_prefix: str, to_user_id: Optional[int]=None, to_email: O
 def send_email_from_dict(email_dict: Mapping[str, Any]) -> None:
     send_email(**dict(email_dict))
 
-def send_future_email(template_prefix: str, realm: Realm, to_user_id: Optional[int]=None,
-                      to_email: Optional[str]=None, from_name: Optional[str]=None,
+def send_future_email(template_prefix: str, realm: Realm, to_user_ids: Optional[List[int]]=None,
+                      to_emails: Optional[List[str]]=None, from_name: Optional[str]=None,
                       from_address: Optional[str]=None, context: Dict[str, Any]={},
                       delay: datetime.timedelta=datetime.timedelta(0)) -> None:
     template_name = template_prefix.split('/')[-1]
-    email_fields = {'template_prefix': template_prefix, 'to_user_id': to_user_id, 'to_email': to_email,
+    email_fields = {'template_prefix': template_prefix, 'to_user_ids': to_user_ids, 'to_emails': to_emails,
                     'from_name': from_name, 'from_address': from_address, 'context': context}
 
     if settings.DEVELOPMENT and not settings.TEST_SUITE:
-        send_email(template_prefix, to_user_id=to_user_id, to_email=to_email, from_name=from_name,
+        send_email(template_prefix, to_user_ids=to_user_ids, to_emails=to_emails, from_name=from_name,
                    from_address=from_address, context=context)
         # For logging the email
 
-    assert (to_user_id is None) ^ (to_email is None)
-    if to_user_id is not None:
+    assert (to_user_ids is None) ^ (to_emails is None)
+    if to_user_ids is not None:
         # The realm is redundant if we have a to_user_id; this assert just
         # expresses that fact
-        assert(UserProfile.objects.filter(id=to_user_id, realm=realm).exists())
-        to_field = {'user_id': to_user_id}  # type: Dict[str, Any]
+        assert(UserProfile.objects.filter(id__in=to_user_ids, realm=realm).exists())
+        to_field = {'user_id': to_user_ids[0]}  # type: Dict[str, Any]
     else:
-        to_field = {'address': parseaddr(to_email)[1]}
+        assert(len(to_emails) >= 1)
+        to_field = {'address': parseaddr(to_emails[0])[1]}
```

Comment:
This logic of dropping all but the first email/user ID is wrong -- the function doesn't behave as documented.  

To avoid my having to re-review the work you've already done, I added a comment at the top of this function warning about this issue, but we should fix it soon in a follow-up commit.